src/server/scripts/Battlegrounds/TempleOfKotmogu/battleground_temple_of_kotmogu.cpp

```diff
@@ -0,0 +1,630 @@
+        sWorldStateMgr->SetValue(TempleOfKotmogu::WorldStates::HordePoints, bg->GetTeamScore(TEAM_HORDE), false, player->GetMap());
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_bg_temple_of_kotmogu_victory_points_periodic::OnHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 112910 - +10 Victory Points!
+template<int32 Points>
+class spell_bg_temple_of_kotmogu_victory_points_kill : public spell_bg_temple_of_kotmogu_victory_points_periodic<Points>
+{
+    void OnHit(SpellEffIndex effIndex) const override
+    {
+        spell_bg_temple_of_kotmogu_victory_points_periodic<Points>::OnHit(effIndex);
+        Player* player = SpellScript::GetHitPlayer();
+        if (!player)
+            return;
+
+        if (ZoneScript* zonescript = player->FindZoneScript())
+            zonescript->DoAction(TempleOfKotmogu::Actions::CheckScore, player, player);
+    }
+
+    void Register() override
+    {
+        SpellScript::OnEffectHitTarget += SpellScript::SpellEffectFn(spell_bg_temple_of_kotmogu_victory_points_kill::OnHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+struct at_bg_temple_of_kotmogu_healing_buff : AreaTriggerAI
+{
+    explicit at_bg_temple_of_kotmogu_healing_buff(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (Player* player = unit->ToPlayer())
+            if (ZoneScript* zonescript = at->GetZoneScript())
+                zonescript->DoAction(TempleOfKotmogu::Actions::ConsumeAreaTrigger, at, player);
+    }
+};
+
+struct at_bg_temple_of_kotmogu_small_area : AreaTriggerAI
+{
+    explicit at_bg_temple_of_kotmogu_small_area(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (Player* player = unit->ToPlayer())
+        {
+            if (!_totalTimeInAreaWithOrb.contains(player->GetGUID()))
+                _totalTimeInAreaWithOrb[player->GetGUID()] = 0;
+
+            player->CastSpell(player, TempleOfKotmogu::Spells::SmallAura, true);
+        }
+    }
+
+    void OnUnitExit(Unit* unit) override
+    {
+        unit->RemoveAurasDueToSpell(TempleOfKotmogu::Spells::SmallAura);
+    }
+
+    void OnUpdate(uint32 diff) override
+    {
+        for (ObjectGuid const& guid : at->GetInsideUnits())
+        {
+            if (guid.IsPlayer())
+            {
+                if (Player* player = ObjectAccessor::FindPlayer(guid))
+                {
+                    if (player->HasAura(TempleOfKotmogu::Spells::PowerOrbImmunityPeriodic))
+                    {
+                        _totalTimeInAreaWithOrb[player->GetGUID()] += diff;
+                        if (_totalTimeInAreaWithOrb[player->GetGUID()] >= 90000 && !player->HasAchieved(TempleOfKotmogu::Achievements::Powerball))
+                            player->CastSpell(player, TempleOfKotmogu::Spells::PowerballCredit, true);
```

Comment:
I know passing true (aka full trigger) is the easiest way, but it's preferable to pass only required cast flags to mimic the behavior if possible.